### PR TITLE
Experiment: visibility of plans on plans page

### PR DESCRIPTION
**Pull request by aneeshd16**
Related to https://github.com/Automattic/martech/issues/3119

## Proposed Changes

* Implements the changes required for the experiment (22038-explat-experiment). 

## Why are these changes being made?

* Please see pau2Xa-6n5-p2

## Testing Instructions

* Assign yourself to the treatment version by going to 22038-explat-experiment
* Go to `/plans/<site slug>` for:
  * a site on a free plan: Confirm that there is no change from production.
  * a site on a Personal plan: Confirm the Personal plan is shown as a spotlight plan and the Free plan is hidden in the grid.
  * a site on a Premium plan: Confirm the Premium plan is shown as a spotlight plan and the Free & Personal plans are hidden in the grid.
  * a site on a Business plan: Confirm the Business plan is **not shown** as a spotlight plan, and the Free, Personal and Premium plans are hidden in the grid.
  * a site on a Commerce plan: Confirm the Commerce plan is **not shown** as a spotlight plan, and the Free, Personal, Premium and Business plans are hidden in the grid.

For screenshots, please see pbxNRc-4gN-p2

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/plans-features-main/index.tsx [hunk 1]
```diff
@@ -820,7 +843,12 @@ const PlansFeaturesMain = ( {
 										coupon={ coupon }
 										currentSitePlanSlug={ sitePlanSlug }
 										generatedWPComSubdomain={ resolvedSubdomainName }
-										gridPlanForSpotlight={ gridPlanForSpotlight }
+										gridPlanForSpotlight={
+											gridPlansForFeaturesGrid.length >= 4 ||
+											hideLowerTierPlansExperimentAssignment?.variationName !== 'treatment'
+												? gridPlanForSpotlight
+												: undefined
+										}
```
Comment (aneeshd16) [inline client/my-sites/plans-features-main/index.tsx:846]: * If control, pass `gridPlanForSpotlight`.
* If treatment and `gridPlansForFeaturesGrid.length < 4`, pass `gridPlanForSpotlight`.
* If treatment and `gridPlansForFeaturesGrid.length >= 4`, pass `undefined`.

File: packages/plans-grid-next/src/components/features-grid/index.tsx [hunk 2]
```diff
@@ -260,7 +260,7 @@ const TabletView = ( {
 	const gridPlansWithoutSpotlight = ! gridPlanForSpotlight
 		? renderedGridPlans
 		: renderedGridPlans.filter( ( { planSlug } ) => gridPlanForSpotlight.planSlug !== planSlug );
-	const numberOfPlansToShowOnTop = 4 === gridPlansWithoutSpotlight.length ? 2 : 3;
+	const numberOfPlansToShowOnTop = 4 === gridPlansWithoutSpotlight.length ? 4 : 3;
```
Comment (aneeshd16) [inline packages/plans-grid-next/src/components/features-grid/index.tsx:263]: If there are only 4 plans, they can be shown on a single row without issues.

https://github.com/user-attachments/assets/9a93807c-d537-4fe3-9bbe-11e88db243bc

**Comment by jeyip** [comment]
## Testing
### Requirements

For each Calypso admin plans page:
- [ ] Verified header price and billing timeframe price against production
- [ ] Verified pricing for monthly and yearly pricing
- [ ] Ensured that a plan could be purchased
- [ ] Spot checked desktop and mobile against production
- [ ] Looked at pro rated pricing
- [ ] Spot checked the spotlight plan

### Browsers
- [x] Chrome

**Review by oswian — COMMENTED** [review]
> Go to /plans/<site slug> for:
> - a site on a free plan: Confirm that there is no change from production.
> - a site on a Personal plan: Confirm the Personal plan is shown as a spotlight plan and the Free plan is hidden in the grid.
> - a site on a Premium plan: Confirm the Premium plan is shown as a spotlight plan and the Free & Personal plans are hidden in the grid.
> - a site on a Business plan: Confirm the Business plan is not shown as a spotlight plan, and the Free, Personal and Premium plans are hidden in the grid.
> - a site on a Commerce plan: Confirm the Commerce plan is not shown as a spotlight plan, and the Free, Personal, Premium and Business plans are hidden in the grid.
> 

✅  Tested each of these plan types, works as described.

**Comment by oswian** [comment]
We have another plans grid experiment running at the moment: 21954-explat-experiment. Was the plan to launch this experiment after that one has completed?

**Comment by jeyip** [comment]
@oswian yes! This is something that @howarding mentioned in our experiment p2 post pbxNRc-4gN-p2#comment-6130 This experiment _would_ interact with the "simplify plans features" experiment you're running, so the plan is to merge the "hide lower tier plans" experiment after yours runs to completion.

**Review by necronet — APPROVED** [review]
ExPlat: Approved

**Pull request merged**